### ISSUE TITLE
No any IntelliSense while editing csproj file of .net core project.

### ISSUE THREAD

**Issue by Xiu-Li** [microsoft/msbuild#2787]
### Steps to reproduce
- Create a new .net core/standard project.
- Right-click project, select "Edit *.csproj".
- No any IntelliSense while editing csproj file.
![image](https://user-images.githubusercontent.com/24784908/33756845-4b1bc91e-dc32-11e7-911a-659f0958383a.png)
### Environment data
VS Build: d15.6stg.27207.3002
dotnet --info:
![image](https://user-images.githubusercontent.com/24784908/33757032-44e1a4fa-dc33-11e7-8a93-cb2d1376219a.png)

**Comment by kkm000** [comment]
I always add `xmlns="http://schemas.microsoft.com/developer/msbuild/2003"` to the main document `Project` node, try this as a workaround. Makes `/pp:` output much cleaner too!

**Comment by Xiu-Li** [comment]
Validate this on d15.6/27401.00 (15.6 Preview 4):
For C# and VB .Net Core project, the IntelliSense can work well.
But for F# .Net Core project, the IntelliSense can't work well.
![image](https://user-images.githubusercontent.com/24784908/35723634-f769237c-0835-11e8-97c2-cbee8a088d53.png)

**Comment by kkm000** [comment]
@Xiu-Li, again, simply add `xmlns="http://schemas.microsoft.com/developer/msbuild/2003` attribute, like <Project Sdk="..." xmlns="http://schemas.microsoft.com/developer/msbuild/2003">. This does not affect the build, but makes the XML editor pick up the schema, and enables Intellisense.

In any case, the magic of referring the namespace happens in some VS extension, I doubt the MsBuild repository is the correct place to report the problem. Sorry, I cannot suggest where is the right repository for that. What happens for C# projects is the XML editor extension in Visual Studio is hooked up by some other extension, which supplies the namespace when sees a namespace-less document with the `.csproj` extension. Try asking the roslyn team.

**Comment by Xiu-Li** [comment]
@kkm000, thanks very much, after adding `xmlns="http://schemas.microsoft.com/developer/msbuild/2003 attribute`, enables Intellisense successfully.

**Comment by couven92** [comment]
I have this issue to. **But adding `xmlns="http://schemas.microsoft.com/developer/msbuild/2003"` to the project node does NOT resolve the issue for me!**

**Comment by tom-corwin** [comment]
I had the same issue. Adding the schema worked for me after restarting Visual Studio.

**Comment by couven92** [comment]
I just rechecked, unlike @tom-corwin I really do not get any IntelliSense for my csproj file. Adding the schema does not work, and after restarting Visual Studio it still does not work. After completely reinstalling Visual Studio it still does not work.

What is **VERY** strange however, is that IntelliSense in the `Directory.Build.props` works flawlessly!

**Comment by tom-corwin** [comment]
@couven92 I'm not sure if it will help. But in the 'XML>Schemas...' menu, have you tried removing and readding the local copy of the schema?

**Comment by couven92** [comment]
@tom-corwin nope, does not help, still no IntelliSense.

**Comment by EnricoReich** [comment]
I have the same problem with .Net Core Projects. IntelliSense for .csproj files isn't working.
BUT IntelliSense works if you unload the project and then edit the .csproj file.

**Comment by YunWeiaa** [comment]
Yes, no repro on VS2017 Enterprise 15.8.3.

**Comment by gorangrubic** [comment]
+EnrichoReich
The problem persists  (multitarget, .NET Standard 2.0), even after:
- adding the schema 
- unloading the project and opening .cproj
VS Community 2017 15.7.4

**Comment by EnricoReich** [comment]
Hey,

I did some investigations and found out, if you put your project into a solution folder like

![brokenintellisense](https://user-images.githubusercontent.com/316637/45843530-2a3c5400-bd20-11e8-8feb-5a60ebc4ec45.png)

Intellisense does not work properly after reloading the solution. This also happens with .Net-Framework projects.

So, this is sad because I like to organize my projects into solution folders.  :disappointed:

**Comment by kkm000** [comment]
@EnricoReich: Looks like it, thanks for nailing it down!  Unloading a project before editing (i. e. not the shadow-editing with the project still loaded) does not pick up the IS schema either, even though all the right schemata are checked as "use" in the file properties window. But if you copy the file, give it an extension ".xml" and open, then IS works. As if the schema is forced into the XML editor by some Roslyn VS integration component, when the VS is aware (let's say, *way too* aware in this case) that  the file is actually a project file, not just any run-off-the-mill XML document.

If only we find the *right* team to deal with the issue. I am sure this is not a problem with MSBuild! Let me try to find the folk on GitHub who likely own that part.

**Comment by tb-mtg** [comment]
Problem still happening for me too at version VS Community 2017 15.8.6

**Comment by tom-corwin** [comment]
@kkm000, I know there's a repo for [ASP.NET Core tooling](https://github.com/aspnet/tooling) for VS, but I don't know if there is one just for .NET Core. If there is, that's probably the best place for this issue (i'd assume, anyways)

**Comment by EnricoReich** [comment]
I described this problem a few weeks ago at the Visual Studio Developer Community website but apparently there is hardly anyone interested in it. Maybe if it gets some upvotes someone will take care of it.

https://developercommunity.visualstudio.com/content/problem/340537/intellisense-for-csproj-file-not-working-if-projec.html

**Comment by tom-corwin** [comment]
@EnricoReich I gave it an upvote, hopefully they'll fix it sooner rather than later.

**Comment by tb-mtg** [comment]
@tom-corwin  @EnricoReich I also gave it an upvote.

**Comment by davkean** [comment]
Here's another two hits of this: 

https://github.com/dotnet/project-system/issues/3225
https://github.com/dotnet/project-system/issues/4061#issuecomment-434599079

**Comment by YunWeiaa** [comment]
Not repro on VS-Val/d15.9/28230.54 (15.9 preview 5.0). For C#/VB/F# .Net Core project, the IntelliSense can work well.

**Comment by huoyaoyuan** [comment]
For my cases:
OS: Win10 Pro 17763, Simplified Chinese
VS: 15.8.8, Simplified Chinese/English

When switching VS to English, IntelliSense works with a simple playground project. When switching to Chinese it does not work.
But it does not work with my [customized projects](https://github.com/amatukaze/ing/blob/master/src/Infrastructure/Sakuno.ING.Standard/Sakuno.ING.Standard.csproj), even in English.
Further information: these projects use `Directory.Build.props` to import target framework, and VS cannot load them if explicitly set project GUID in solution file.

**Comment by EnricoReich** [comment]
I can repro this in VS-Enterprise 15.8.9 & 15.9preview5.

OS: Windows 10 Pro [17763.55, 1809], Language: German
VS: 15.8.9 & 15.9preview5 (without additional extensions), Language: English (both)

If I create a project at solution root (in VS) Intellisense for csproj files works as intended. But if I create or drag&drop a project into a solution folder/directory and reload the solution, it does not work e.g. almost no Intellisense for csproj files. Only code completions for CDATA and comments are shown.

**Comment by couven92** [comment]
Same here as @EnricoReich describes. I actually use a `Directory.props` file to enforce the following .NET repository structure:

``` txt
root
├─ bin
|  ├─ Debug
|  |  ├─ netstandard1.3
|  |  └─ netstandard2.0
|  └─ Release
|     ├─ netstandard1.3
|     └─ netstandard2.0
├─ obj
|  └─ ...
├─ src
|  └─ ...
├─ sample
|  └─ ...
├─ test
|  └─ ...*.Test
├─ .gitignore
├─ .gitattributes
├─ .editorconfig
├─ LICENSE
├─ README.md
├─ AllRules.ruleset
├─ Directory.Build.props
└─ *.sln
```

I use `dotnet sln add` to add projects to the solution which will organise the projects in solution folders that match the actual on-disk file system hierarchy.

**Comment by robertmclaws** [comment]
Just an FYI, even though I get limited Intellisense in the latest VS 15.9 builds, it still doesn't have any information on `None`/`Content` attributes like `Pack` and `PackagePath`.

**Comment by ScotBren** [comment]
I don't yet know the underlying cause of the problem, but let me share the experience I just had with a customer reporting the same problem in the VS 2019 Enterprise 16.4.5 XML editor, trying to edit a .csproj. Setting the xmlns made no difference to him.  
However, when I observed that his system was set to locale 1036 (French - France) and mine is 1033 (English - US), just as an experiment, I zipped up all of my xml\schemas folder and subfolders and shared them with him.  He dropped that into his machine, and the IntelliSense is working for him.  We haven't yet taken the time to investigate the differences, but it's important to note that this problem may be related to language/locale-specific differences.

**Comment by rainersigwald** [comment]
It looks like there may be multiple issues reported here. I think the main one was fixed as #2723 for Visual Studio 16.5. I'm going to close this issue, but please don't hesitate to file a new issue if you see problems on 16.5 or higher. Please include details like your OS and Visual Studio versions, language settings, and what you expect to see and actually experience.

**Comment by Legends** [comment]
VS 2019: I get intelli for the `import `element but not for the attributes. The `import `is located directly under `Projects`
